src/editors/containers/ProblemEditor/data/ReactStateOLXParser.js

```diff
@@ -96,42 +94,91 @@ class ReactStateOLXParser {
     return compoundhint;
   }
 
-  addQuestion() {
+  parseQuestionFromState() {
     const { question } = this.problemState;
-    const questionObject = this.parser.parse(question);
+    const questionObject = new XMLParser({
+        ...this.parserOptions,
+        preserveOrder: true,
+        trimValues: false,
+      }).parse(question);
     return questionObject;
   }
 
+  insertQuestionIntoOLX(OLX, problemType, indentationLevel) {
+    // it's necessary to build the question with the different options, but
+    // this options will mess with the rest of the generated OLX, so we build
+    // them separately, and merge two strings; hacky, but works nicely without
+    // messing up formatting of the question or the rest of the XML
+    const question = this.parseQuestionFromState();
+    const questionBuild = new XMLBuilder({
+      ...this.builderOptions,
+      preserveOrder: true,
+      format: false,
+    }).build(question);
```

Comment:
@Cup0fCoffee 

The question in `problemState`
```
const { question } = this.problemState;
```
already seems to be stored as an XML. Here we are parsing the question XML and then re-building it. I tested the code briefly, and it seemed to me that we would simply use the question XML as is from `problemState`. Am I missing something here ?

Reply:
@kaustavb12 You're right. I was wrongfully thinking that parsing and building it would add extra guarantees in terms of consistency of the formatting, but it's really just redundant. Removing it.